### DISCUSSION TITLE
Only poll when element is visible

### DISCUSSION THREAD

**Discussion by joshuadegier** [livewire/livewire#2233] [3. Feature ideas]
Hi there!

We've been using Livewire quiet some time now and while options seem endless, there's this one little thing that we've recently found to be basically impossible. Or at least ... for us. What we're trying to do is pretty straight forward: to halt polling whenever an element is not visible on the page.

We've looked at a few work arounds such as setting the `wire:poll` attribute via php, but we cannot link anything from the Livewire component to the knowledge of it's visbility (all JS). Then we tried to hook into other events removing and restoring the `wire:poll` attribute based on scrolling/sliding on the page. But.. that led to some weird behaviour in random polling requests (which still, of course, could be our fault).

Wether or not it's a lack of knowledge of the options on our side, this still made me wonder.. wouldn't it be a nice feature to be able to do something like this?

`wire:poll-visible.1000ms`

Our use case to look into this option is pretty simple: data is being fetched from an external API and while we have a caching layer in place it's still unneeded extra API requests when they happen in the background in between the caching intervals.

**Comment by joshuadegier** [comment]
I'm willing to add a EUR 250 bounty to whoever is able to find a solution for this idea!

**Comment by boynet** [comment]
cant you implement it yourself?
* detect when element is visible
* while he is visible call refresh\event every X times
dont use the poll method at all

**Comment by ju5t** [comment]
@joshuadegier I've opened up a PR for this at https://github.com/livewire/livewire/pull/2372. It does exactly what you're after.
